src/components/FeatureExportButton/FeatureExportButton.js

```diff
@@ -0,0 +1,209 @@
+import Style from 'ol/style/Style';
+import Circle from 'ol/style/Circle';
+import Icon from 'ol/style/Icon';
+import VectorLayer from '../../VectorLayer';
+import Button from '../Button';
+
+import { kmlStyle } from '../../utils/Styles';
+
+const propTypes = {
+  /**
+   *  Children content of the Feature export button.
+   */
+  children: PropTypes.element,
+
+  /**
+   * CSS class of the Feature export button.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Format to export features (function).
+   */
+  format: PropTypes.func,
+
+  /** An existing react-spatial VectorLayer, using a valid ol.source.Vector */
+  layer: PropTypes.instanceOf(VectorLayer).isRequired,
+
+  /**
+   * Map projection.
+   */
+  projection: PropTypes.string,
+
+  /**
+   * HTML tabIndex attribute
+   */
+  tabIndex: PropTypes.number,
+
+  /**
+   * Title of the Feature export button.
+   */
+  title: PropTypes.string,
+};
+
+const defaultProps = {
+  children: <FaCloudDownloadAlt focusable={false} />,
+  className: 'tm-feature-export',
+  format: KML,
+  projection: 'EPSG:3857',
+  tabIndex: 0,
+  title: undefined,
+};
+
+/**
+ * This component displays a button to export geometry feature.
+ */
+class FeatureExportButton extends PureComponent {
+  createFeatureString(layer) {
+    const { projection, format } = this.props;
+    const { olLayer } = layer;
+
+    let featString;
+    const exportFeatures = [];
+
+    olLayer.getSource().forEachFeature(f => {
+      // We silently ignore Circle elements as they are
+      // not supported in kml
+      if (f.getGeometry().getType() === 'Circle') {
+        return;
+      }
+
+      const clone = f.clone();
+      clone.setId(f.getId());
+      clone.getGeometry().setProperties(f.getGeometry().getProperties());
+      clone.getGeometry().transform(projection, 'EPSG:4326');
+
+      // Extended data breaks KML validty so we remove it for now.
+      for (let i = 0; i < f.getProperties(); i += 1) {
+        if (/^(geometry|name|description)$/.test(i)) {
+          // eslint-disable-next-line no-continue
+          continue;
+        }
+        clone.unset(i, true);
+      }
+
+      let styles;
+
+      if (clone.getStyleFunction()) {
+        styles = clone.getStyleFunction().call(clone);
+      } else {
+        styles = olLayer.getStyleFunction().call(olLayer, clone);
+      }
+      const newStyle = {
+        fill: styles[0].getFill(),
+        stroke: styles[0].getStroke(),
+        text: styles[0].getText(),
+        image: styles[0].getImage(),
+        zIndex: styles[0].getZIndex(),
+      };
+
+      if (newStyle.image instanceof Circle) {
+        newStyle.image = null;
+      }
+
+      // If only text is displayed we must specify an
+      // image style with scale=0
+      if (newStyle.text && !newStyle.image) {
+        newStyle.image = new Icon({
+          src: 'noimage',
+          scale: 0,
+        });
+      }
+
+      const myStyle = new Style(newStyle);
+      clone.setStyle(myStyle);
```

Comment:
newStyle is already new no need to create a another one.

Reply:
need to instantiate it as ol style here